python_language/inventory/inventory.py

```diff
@@ -0,0 +1,46 @@
+"""
+Inventory
+---------
+
+Calculate and report the current inventory in a warehouse.
+
+Assume the warehouse is initially empty.
+
+The string, warehouse_log, is a stream of deliveries to
+and shipments from a warehouse.  Each line represents
+a single transaction for a part with the number of
+parts delivered or shipped.  It has the form::
+
+    part_id count
+
+If "count" is positive, then it is a delivery to the
+warehouse. If it is negative, it is a shipment from
+the warehouse.
+
+See :ref:`inventory-solution`.
+"""
+
+warehouse_log = """ frombicator      10
+                    whitzlegidget    5
+                    splatzleblock    12
+                    frombicator      -3
+                    frombicator      20
+                    foozalator       40
+                    whitzlegidget    -4
+                    splatzleblock    -8
+                """
+                
+warehouse_log=warehouse_log.strip()
+list_log=warehouse_log.split('\n')
+log={}
```

Comment:
I like to encourage use of names that reflect what you're working with.  `log` here implies that it's some kind of record of transactions, when in fact, I think you are tracking inventory.  I would recommend changing the name to `inventory` or `warehouse` or similar.